crates/wasmtime/src/runtime/vm/gc/gc_runtime.rs

```diff
@@ -356,45 +382,39 @@ pub unsafe trait GcHeap: 'static + Send + Sync {
     unsafe fn vmctx_gc_heap_data(&self) -> NonNull<u8>;
 
     ////////////////////////////////////////////////////////////////////////////
-    // Recycling GC Heap Methods
+    // Accessors for the raw bytes of the GC heap
 
-    /// Reset this heap.
-    ///
-    /// Calling this method unassociates this heap with the store that it has
-    /// been associated with, making it available to be associated with a new
-    /// heap.
-    ///
-    /// This should refill free lists, reset bump pointers, and etc... as if
-    /// nothing were allocated in this heap (because nothing is allocated in
-    /// this heap anymore).
-    ///
-    /// This should retain any allocated memory from the global allocator and
-    /// any virtual memory mappings.
-    ///
-    /// This method is only used with the pooling allocator.
-    #[cfg(feature = "pooling-allocator")]
-    fn reset(&mut self);
+    /// Take the underlying memory storage out of this GC heap.
+    fn take_memory(&mut self) -> crate::vm::Memory;
 
-    ////////////////////////////////////////////////////////////////////////////
-    // Accessors for the raw bytes of the GC heap
+    /// Replace this GC heap's underlying memory storage.
+    fn replace_memory(&mut self, memory: crate::vm::Memory, delta_bytes_grown: u64);
```

Comment:
This carries an unsafe requirement that the memory being replaced has to be basically the same as `take_memory`, right? In that it must be at least the same size?